tests/test_MOcovMFile_recognizes_classdef_syntax.m

```diff
@@ -0,0 +1,81 @@
+function test_suite = test_MOcovMFile_recognizes_classdef_syntax
+    initTestSuite;
+end
+
+function fullname = tempfile(filename, contents)
+    tempfolder = fullfile(tempdir, 'mocov_fixtures');
+    [~, ~, ~] = mkdir(tempfolder);
+    fullname = fullfile(tempfolder, filename);
+    fid = fopen(fullname, 'w');
+    fprintf(fid, contents);
+    fclose(fid);
+end
+
+function filename = create_classdef
+    filename = tempfile('AClass.m', [ ...
+        'classdef AClass < handle\n', ...
+        '  properties\n', ...
+        '    aProp = 1;\n', ...
+        '  end\n', ...
+        '  properties (SetAccess = private, Dependent)\n', ...
+        '    anotherProp;\n', ...
+        '  end\n', ...
+        '  methods\n', ...
+        '    function self = AClass\n', ...
+        '      fprintf(''hello world!'');\n', ...
+        '    end\n', ...
+        '  end\n', ...
+        '  methods (Access = private)\n', ...
+        '    function x = aMethod(self)\n', ...
+        '      fprintf(''hello world!'');\n', ...
+        '    end\n', ...
+        '  end\n', ...
+        'end\n' ...
+    ]);
+end
+
+function test_classdef_line_not_executable
+    mfile = MOcovMFile(create_classdef);
```

Comment:
The point about having a temporary folder is to be confident about using `addpath` (I don't know what kind of mfiles exist under the temp dir, and maybe them can shadow some other function)...

I guess a way to avoid it is explicitly requiring the folder to be specified in the filename (is this your suggestion in other words?).